static/panes/stack-usage-view.ts

```diff
@@ -0,0 +1,177 @@
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import $ from 'jquery';
+import _ from 'underscore';
+import * as monaco from 'monaco-editor';
+import {Container} from 'golden-layout';
+
+import {MonacoPane} from './pane.js';
+import {StackUsageState, suCodeEntry} from './stack-usage-view.interfaces.js';
+import {MonacoPaneState} from './pane.interfaces.js';
+
+import {ga} from '../analytics.js';
+import {extendConfig} from '../monaco-config.js';
+import {Hub} from '../hub.js';
+
+export class StackUsage extends MonacoPane<monaco.editor.IStandaloneCodeEditor, StackUsageState> {
+    currentDecorations: string[] = [];
+    // Note: bool | undef here instead of just bool because of an issue with field initialization order
+    isCompilerSupported?: boolean;
+
+    constructor(hub: Hub, container: Container, state: StackUsageState & MonacoPaneState) {
+        super(hub, container, state);
+        if (state.suOutput) {
+            this.showStackUsageResults(state.suOutput);
+        }
+        this.eventHub.emit('stackUsageViewOpened', this.compilerInfo.compilerId);
+    }
+
+    override getInitialHTML(): string {
+        return $('#stackusage').html();
+    }
+
+    override createEditor(editorRoot: HTMLElement): monaco.editor.IStandaloneCodeEditor {
+        return monaco.editor.create(
+            editorRoot,
+            extendConfig({
+                language: 'plaintext',
+                readOnly: true,
+                glyphMargin: true,
+            }),
+        );
+    }
+
+    override registerOpeningAnalyticsEvent() {
+        ga.proxy('send', {
+            hitType: 'event',
+            eventCategory: 'OpenViewPane',
+            eventAction: 'StackUsage',
+        });
+    }
+
+    override registerCallbacks() {
+        this.eventHub.emit('requestSettings');
+        this.eventHub.emit('findCompilers');
+
+        this.container.on('shown', this.resize, this);
+
+        const cursorSelectionThrottledFunction = _.throttle(this.onDidChangeCursorSelection.bind(this), 500);
+        this.editor.onDidChangeCursorSelection(e => {
+            cursorSelectionThrottledFunction(e);
+        });
+    }
+
+    override onCompileResult(id: number, compiler, result) {
+        if (this.compilerInfo.compilerId !== id || !this.isCompilerSupported) return;
+        this.editor.setValue(result.source);
+        if (result.hasStackUsageOutput) {
+            this.showStackUsageResults(result.stackUsageOutput);
+        }
+        // TODO: This is inelegant again. Previously took advantage of fourth argument for the compileResult event.
```

Comment:
Not something you need to take care of in this PR since it's from the opt view but I'm noticing there is now a 4th compileResult argument 🤔